File: Assignment 20/Question4.java
    import java.util.Scanner;

class Logic
{
    
    public int FindMaxDigit(int iNo) 

    {
        int iDigit = 0 ;
        int iMax = 0;

        while (iNo !=0)
        {
            iDigit = iNo % 10;
            iNo = iNo /10 ;
            if (iDigit > iMax)
            {
                iMax = iDigit;
            }  
        }
       return iMax;
    }  
}          
class Question4
{
    public static void main(String[] args) 
    {
        
        Scanner sobj = new Scanner(System.in);
        int iValue1 = 0 , iRet = 0   ;
        
       
        System.out.println("Enter your Number :");
        iValue1 = sobj.nextInt();

        Logic lobj = new Logic();

        iRet = lobj.FindMaxDigit(iValue1);
        System.out.println("The Max Digit is : "+iRet);
        
    }    
}

File: Assignment 15/Question4.c
#include <stdio.h>


int MultiplyDigit(int iNo)
{
    int iDigit = 0;
    int iMultiplication = 0;
    iMultiplication = 1;

    if (iNo < 0)
    {
        iNo = -iNo ;
    }
    while(iNo !=0)
    {
        iDigit = iNo % 10;
        iNo = iNo/10 ;
        if(iDigit ==0)
        {
            iDigit =1;
        }
        iMultiplication = iMultiplication * iDigit;
     }
     return iMultiplication; 

}

int main()
{
    int iValue = 0 ;
    int iRet  = 0 ;
    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = MultiplyDigit(iValue);

     
    printf("%d is the Multiplication ",iRet);
    
    return 0;
}